cryptoconditions/crypto.py

```diff
@@ -1,269 +1,121 @@
-
-        Args:
-            prefix:
-            encoding:
-        """
+    def encode(data):
+        return base58.b58encode(data).encode()
 
     @staticmethod
-    @abstractmethod
-    def encode(public_value):
-        """
-        Encode the public key to base58 represented by the internal values
-
-        Args:
-            public_value
-        """
-
-    @staticmethod
-    @abstractmethod
-    def decode(public_base58):
-        """
-        Decode the base58 public_value to internal value
-
-        Args:
-            public_base58 (base58):
-        """
-
-
-class Ed25519SigningKey(ed25519.SigningKey, SigningKey):
+    def decode(data):
+        return base58.b58decode(data)
+
+
+def _get_nacl_encoder(encoding):
+    if encoding == 'base58':
+        return Base58Encoder
+    elif encoding == 'base64':
+        return nacl.encoding.Base64Encoder
+    elif encoding == 'base32':
+        return nacl.encoding.Base32Encoder
+    elif encoding == 'base16':
+        return nacl.encoding.Base16Encoder
+    elif encoding == 'hex':
+        return nacl.encoding.HexEncoder
+    elif encoding is 'bytes':
+        return nacl.encoding.RawEncoder
+    else:
+        raise exceptions.UnknownEncodingError("Unknown or unsupported encoding")
+
+
+class Ed25519SigningKey(nacl.signing.SigningKey):
     """
     PrivateKey instance
     """
 
-    def __init__(self, key):
+    def __init__(self, key, encoding='base58'):
         """
         Instantiate the private key with the private_value encoded in base58
 
         Args:
             key (base58): base58 encoded private key
         """
-        private_base64 = self.decode(key)
-        super().__init__(private_base64, encoding='base64')
+        super().__init__(key, encoder=_get_nacl_encoder(encoding))
 
     def get_verifying_key(self):
         """
         Get the corresponding VerifyingKey
 
         Returns:
-            VerifyingKey
+            Ed25519VerifyingKey
         """
-        vk = super().get_verifying_key()
-        return Ed25519VerifyingKey(base58.b58encode(vk.to_bytes()))
+        return Ed25519VerifyingKey(self.verify_key.encode(encoder=Base58Encoder))
 
-    def to_ascii(self, prefix="", encoding='base58'):
-        """
-        convert external value to ascii with specified encoding
-
-        Args:
-            prefix (str):
-            encoding (str): {'base58'|'base64'|'base32'|'base16'|'hex'}
-
-        Returns:
-            bytes: encoded string
-        """
-        if encoding == 'base58':
-            return base58.b58encode(self.to_seed()).encode('ascii').decode('ascii').rstrip("=").encode('ascii')
-        else:
-            return super().to_ascii(prefix=prefix, encoding=encoding)
-
-    def sign(self, data, prefix="", encoding="base58"):
+    def sign(self, data, encoding="base58"):
         """
         Sign data with private key
 
         Args:
             data (str, bytes): data to sign
-            prefix:
             encoding (str): base64, hex
```

Comment:
Probably should be `encoding (str): {'base58'|'base64'|'base32'|'base16'|'hex'}`?